jar_test.go

```diff
@@ -1964,6 +1964,77 @@ func TestRemoveCookies(t *testing.T) {
 	}
 }
 
+func TestRemoveAll(t *testing.T) {
+	testRemoveAll(t, mustParseURL("https://www.apple.com"), "www.apple.com")
+}
+
+func TestRemoveAllRoot(t *testing.T) {
+	testRemoveAll(t, mustParseURL("https://www.apple.com"), "apple.com")
```

Comment:
Yep.. fixed and added a test for different subdomains *not* removing each other.